packages/atlas/src/views/studio/PlaylistWorkspace/VideoSelectorDialog/VideoSelectorDialog.tsx

```diff
@@ -0,0 +1,111 @@
+import { Text } from '@/components/Text'
+import { Button } from '@/components/_buttons/Button'
+import { VideoListItem, VideoListItemLoader } from '@/components/_video/VideoListItem'
+import { useDebounceValue } from '@/hooks/useDebounceValue'
+import { useAuthorizedUser } from '@/providers/user/user.hooks'
+import { createId } from '@/utils/createId'
+import { SentryLogger } from '@/utils/logs'
+
+import { ListWrapper, StyledDialogModal, StyledInput, Wrapper } from './VideoSelectorDialog.styles'
+
+type VideoSelectorDialogProps = {
+  onHide: () => void
+  show: boolean
+  onSelect: (arg: [string, string][]) => void
+}
+const INITIAL_FIRST = 50
+
+export const VideoSelectorDialog: FC<VideoSelectorDialogProps> = ({ onHide, show, onSelect }) => {
+  const [selectedVideoIds, setSelectedVideoIds] = useState<string[]>([])
+  const [search, setSearch] = useState<string>('')
+  const { channelId } = useAuthorizedUser()
+  const debouncedSearch = useDebounceValue(search, 200)
+  const { edges, loading } = useFullVideosConnection(
+    {
+      first: INITIAL_FIRST,
+      orderBy: VideoOrderByInput.CreatedAtDesc,
+      where: {
+        channel: {
+          id_eq: channelId,
+        },
+        title_contains: debouncedSearch,
+      },
+    },
+    {
+      notifyOnNetworkStatusChange: true,
+      onError: (error) => SentryLogger.error('Failed to fetch videos', 'MyVideosView', error),
+    }
+  )
+
+  return (
+    <StyledDialogModal
+      show={show}
+      size="medium"
+      contentClassName="dialog-content"
+      primaryButton={{
+        text: 'Add videos',
+        disabled: !selectedVideoIds.length,
+        onClick: () => {
+          onSelect(selectedVideoIds.map((videoId) => [videoId, createId()]))
+          setSelectedVideoIds([])
+          onHide()
+        },
+      }}
+      secondaryButton={{ text: 'Cancel', onClick: onHide }}
+      additionalActionsNode={
+        <>
+          <Button
+            variant="tertiary"
+            onClick={() => edges?.length && setSelectedVideoIds(edges.map((edge) => edge.node.id))}
+          >
+            Select all
+          </Button>
+          {!!selectedVideoIds.length && (
+            <Text as="p" variant="t300" color="colorTextMuted">
+              {selectedVideoIds.length} video(s) selected
+            </Text>
+          )}
+        </>
+      }
+      actionDivider
+    >
+      <Wrapper>
+        <StyledInput
+          value={search}
+          onChange={(e) => setSearch(e.target.value)}
+          nodeStart={<SvgActionSearch />}
+          placeholder="Search videos to add"
+        />
+        {loading ? (
+          <ListWrapper>
+            {Array.from({ length: 8 }, (_, idx) => (
+              <VideoListItemLoader key={idx} variant="small" />
+            ))}
+          </ListWrapper>
+        ) : (
+          <ListWrapper>
+            {edges?.length
+              ? edges.map(({ node: { id } }) => (
+                  <VideoListItem
```

Comment:
I think I said this already, there is no hover effect on the list item. Designers use here ListItem component 
![Zrzut ekranu 2023-01-16 o 19 41 58](https://user-images.githubusercontent.com/51168865/212747137-dd87e3a3-699f-4728-871a-7ecfc7934285.png)
And if you inspect ListItem, you'll notice that this component has a hover effect defined: 
![image](https://user-images.githubusercontent.com/51168865/212746839-d6d55b40-a617-4f08-9254-976c2bae5d18.png)

Also, can we use our component here? https://atlas-storybook-git-dev-joystream.vercel.app/?path=/story/other-listitem--default